aicsimageio/readers/tiff_reader.py

```diff
@@ -175,48 +193,88 @@ def _create_dask_array(self) -> da.Array:
+                        non_chunk_shape.append(size)
+
+                # Fill out the rest of the blocked shape with dimension sizes of 1 to
+                # match the length of the sample chunk
+                # When dask.block happens it fills the dimensions from inner-most to
+                # outer-most with the chunks as long as the dimension is size 1
+                blocked_dim_order = non_chunk_dim_order + chunk_dim_order
+                blocked_shape = tuple(non_chunk_shape) + ((1,) * len(chunk_shape))
 
                 # Make ndarray for lazy arrays to fill
-                lazy_arrays = np.ndarray(operating_shape, dtype=object)
+                lazy_arrays = np.ndarray(blocked_shape, dtype=object)
                 for plane_index, (np_index, _) in enumerate(
                     np.ndenumerate(lazy_arrays)
                 ):
+                    # All dimensions get their normal index except for chunk dims
+                    # which get filled with "full" slices
+                    indices_with_slices = np_index[: len(non_chunk_shape)] + (
+                        (slice(None, None, None),) * len(chunk_shape)
+                    )
+
                     # Fill the numpy array with the delayed arrays
                     lazy_arrays[np_index] = da.from_delayed(
                         delayed(TiffReader._get_image_data)(
                             fs=self.fs,
                             path=self.path,
                             scene=self.current_scene_index,
-                            index=plane_index,
+                            indices=indices_with_slices,
                         ),
-                        shape=sample.shape,
-                        dtype=sample.dtype,
+                        shape=chunk_shape,
+                        dtype=selected_scene.dtype,
                     )
 
                 # Convert the numpy array of lazy readers into a dask array
                 image_data = da.block(lazy_arrays.tolist())
 
+                # Because we have set certain dimensions to be chunked and others not
+                # we will need to transpose back to original dimension ordering
+                # Example, if the original dimension ordering was "TZYX" and we
+                # chunked by "T", "Y", and "X"
+                # we created an array with dimensions ordering "ZTYX"
+                transpose_indices = []
+                transpose_required = False
+                for i, d in enumerate(selected_scene_dims):
+                    new_index = blocked_dim_order.index(d)
+                    if new_index != i:
+                        transpose_required = True
+                        transpose_indices.append(new_index)
+                    else:
+                        transpose_indices.append(i)
+
+                # Only run if the transpose is actually required
+                if transpose_required:
+                    image_data = da.transpose(image_data, tuple(transpose_indices))
```

Comment:
This is not essential, but it just occurred to me that an identity transpose should be really cheap or no-op.  So you could remove all the conditional stuff here and just build up a transpose that either does or does not shuffle dim indices.

Reply:
Won't resolve now. May resolve in the future if I benchmark and see no real diff.